WeakAuras/WeakAuras.lua

```diff
@@ -1470,8 +1475,9 @@ function WeakAuras.ScanForLoads(self, event, arg1)
   for id, data in pairs(db.displays) do
     if (data and not data.controlledChildren) then
       local loadFunc = loadFuncs[id];
-      shouldBeLoaded = loadFunc and loadFunc("ScanForLoads_Auras", incombat, vehicle, vehicleUi, inpetbattle, group, player, realm, class, spec, race, faction, playerLevel, zone, zoneId, zonegroupId, encounter_id, size, difficulty, role);
-      couldBeLoaded = loadFunc and loadFunc("ScanForLoads_Auras", true, vehicle, vehicleUi, true, group, player, realm, class, spec, race, faction, playerLevel, zone, zoneId, zonegroupId, encounter_id, size, difficulty, role);
+      local loadOpt = loadFuncsForOptions[id];
+      shouldBeLoaded = loadFunc and loadFunc("ScanForLoads_Auras", incombat, inencounter, inpetbattle, vehicle, vehicleUi, group, player, realm, class, spec, race, faction, playerLevel, zone, zoneId, zonegroupId, encounter_id, size, difficulty, role);
+      couldBeLoaded =  loadOpt and loadOpt("ScanForLoads_Auras",   nil,      nil,         nil,         nil,     nil,       nil,   player, realm, class, spec, race, faction, playerLevel, nil,  nil,    nil,         nil,          nil,  nil,        role);
```

Comment:
I’m not sure if anything can be reasonably done to improve this, but I don’t really like the idea of passing a dozen nils to a function.

Reply:
I did that on purpose, to indicate which values are ignored by loadOpt

Reply:
I get that, I just wonder if there isn’t a more elegant solution. I suppose there isn’t without redesigning how ScanForLoads_Auras works. It’s fine as it is.